hack/test-extended.sh

```diff
@@ -91,16 +94,26 @@ start_server() {
 
 start_docker_registry() {
   mkdir -p ${BASETMPDIR}/.registry
-  echo "[INFO] Creating Router"
+  echo "[INFO] Creating Router ..."
   openshift ex router --create --credentials="${KUBECONFIG}" \
     --images='openshift/origin-${component}:latest' &>/dev/null
 
-  echo "[INFO] Creating Docker Registry"
+  echo "[INFO] Creating Registry ..."
   openshift ex registry --create --credentials="${KUBECONFIG}" \
     --mount-host="${BASETMPDIR}/.registry" \
     --images='openshift/origin-${component}:latest' &>/dev/null
 }
 
+pull_and_push() {
+  local image=$1
+  local registry=$2
+  echo "[INFO] Caching $image to $registry"
+  ( docker pull $image && \
```

Comment:
I'm not sure we want to do this pull because IIRC we build those images locally when doing full test inside jenkins. So this might overwrite them, which is not what we'd expect. The same might be true for local as well. In all other cases where you'd actually wanted to pull them I'd would add flag to those tests to do so.